pkg/controller/plan/kubevirt.go

```diff
@@ -693,8 +693,17 @@ func (r *KubeVirt) getStorageProfileModes(storageName string) (accessModes []cor
 		storageProfile := &storageProfileList.Items[i]
 		if storageProfile.Name == storageName {
 			for _, claimProperty := range storageProfile.Status.ClaimPropertySets {
-				accessModes = append(accessModes, claimProperty.AccessModes...)
 				volumeMode = claimProperty.VolumeMode
+				// Preferring Block volume mode
+				if *claimProperty.VolumeMode == core.PersistentVolumeBlock && *volumeMode != core.PersistentVolumeBlock {
+					volumeMode = claimProperty.VolumeMode
+					break
+				}
```

Comment:
isn't it over complicated? why not to just check if *volumeMode == core.PersistentVolumeBlock after the assignment and then break?

Reply:
yes, done

Reply:
would have been nicer to have:
```
if volumeMode = claimProperty.VolumeMode; *volumeMode == core.PersistentVolumeBlock {
        // Preferring Block volume mode
	break
}
```
in my opinion but that's a matter of taste